mux_test.go

```diff
@@ -359,6 +359,58 @@ func TestMuxNestedNotFound(t *testing.T) {
 	}
 }
 
+func TestMuxComplicatedNotFound(t *testing.T) {
+	// sub router with groups
+	sub := NewRouter()
+	sub.Route("/resource", func(r Router) {
+		r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte("private get"))
+		})
+	})
+
+	// Root router with groups
+	r := NewRouter()
+	r.Get("/auth", func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("auth get"))
+	})
+	r.Route("/public", func(r Router) {
+		r.Get("/", func(w http.ResponseWriter, r *http.Request) {
+			w.Write([]byte("public get"))
+		})
+	})
+	r.Mount("/private", sub)
+	r.NotFound(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
yea, I do like the idea that NotFound would be used all the way down from this point in the tree. If another subrouter defined NotFound it would use that one all the way down. Nice work.

However, what would happen in this test func if r.NotFound() was defined at the very top before the Mount(), I think it might fail